### PR TITLE
feat: Add support for default_secondary_roles

### PR DESCRIPTION
**Pull request by mwiewior**
This to implement feature request #773 
This feature is especially important if you use OAuth2 authentication mechanism with BI tool like Looker due to the
following limitation of this integration:

When using OAuth, you cannot switch to different roles in the Snowflake user account. As described in the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/oauth-custom.html#scope), Snowflake uses the default role of the Snowflake user’s account, unless the default role is ACCOUNTADMIN or SECURITYADMIN. Because these roles are blocked for OAuth, Snowflake will instead use the PUBLIC role. See the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/oauth-partner.html#blocking-specific-roles-from-using-the-integration) for information.

## Test Plan

* [ ] acceptance tests

* [ ] …

## References

*

### CODE REVIEW

**Comment by sfc-gh-swinkler** [comment]
hello @mwiewior can you please rebase?

**Comment by mwiewior** [comment]
hello @sfc-gh-swinkler done !

File: pkg/snowflake/user.go [hunk 1]
```diff
@@ -62,6 +65,17 @@ func ScanUserDescription(rows *sqlx.Rows) (*user, error) {
 			r.DefaultNamespace = userProp.Value
 		case "DEFAULT_ROLE":
 			r.DefaultRole = userProp.Value
+		case "DEFAULT_SECONDARY_ROLES":
```
Comment (sfc-gh-swinkler) [inline pkg/snowflake/user.go:68]: i am a bit confused what is happening here. why unmarshal it to a []string only to convert it back to a sql string? do you have a link to any documentation on this?

**Review by sfc-gh-swinkler — COMMENTED** [review]
looks good, but if you could please provide some clarification that would be nice

**Comment by sfc-gh-swinkler** [comment]
also can you please rebase?

File: pkg/snowflake/user.go (hunk 1 above)
Comment (mwiewior) [reply pkg/snowflake/user.go:65]: @sfc-gh-swinkler you're right it wasn't necessary - I refactored it.

File: pkg/resources/user.go [hunk 2]
```diff
@@ -200,6 +207,11 @@ func ReadUser(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
+	err = d.Set("default_secondary_roles", removeEmptyStrings(strings.Split(u.DefaultSecondaryRoles.String, ",")))
```
[the remark this answers is not in the document]
Comment (mwiewior) [reply pkg/resources/user.go]: @sfc-gh-swinkler ok, I refactored this one as well to make it more clear.

**Comment by mwiewior** [comment]
Acceptance tests passed.

**Comment by mwiewior** [comment]
@sfc-gh-swinkler could you pls take a look?

**Comment by sfc-gh-swinkler** [comment]
@mwiewior i am sorry but could you please rebase one more time? I just merged a different PR that that made this one out of date again. Everything else looks good to go for this PR though.

**Comment by mwiewior** [comment]
@sfc-gh-swinkler thx ! I've just rebased.

**Comment by sfc-gh-swinkler** [comment]
@mwiewior thank you for your contribution!

**Comment by mwiewior** [comment]
@sfc-gh-swinkler thx for your valuable comments and help in getting it through!

**Pull request merged**